Issue111 - draft pr

mwilck opened the pull request:
Attempted fix for
https://github.com/opensvc/multipath-tools/issues/111

mwilck committed:
libmpathcmd: try both abstract and pathname sockets

When connecting to the multipathd socket, try the pathname socket
first, then the abstract socket. Fail only if both connection attempts
fail.

Signed-off-by: Martin Wilck <mwilck@suse.com>

mwilck committed:
libmpathcmd: honor MULTIPATH_SOCKET_NAME environment variable

In systemd installments, users can already override the socket names
that systemd listens on. With this patch, clients using libmpathcmd
can be customized to use a non-standard socket by setting an environment
variable.

Signed-off-by: Martin Wilck <mwilck@suse.com>
Reviewed-by: Benjamin Marzinski <bmarzins@redhat.com>

bmarzins reviewed (commented):
When you remove use_regular_socket, you didn't remove the reference in README.md

bmarzins reviewed (commented):
Since we don't compile for just one socket type with this code, we can use your version of mpath_fill_sockaddr__()  that's in its own file, included by both libmpathcmd and multipathd, which avoids needing to export an internal function in libmpathcmd and lets us avoid bumping the version number.

Otherwise, this all looks fine, and it does have the benefit that (assuming we don't have to update the libmpathcmd version number) already compiled programs that access multipathd will continue to just work, which avoids a nagging worry I have that there are people using libmpathcmd whose stuff would stop working and need recompiling when I pulled this change in.

mwilck commented:
OK, thank you. I was aware of the README issue. I was undecided wrt the `libmpathcmd.so` vs. `#include` question but I suppose you're right, there is no incompatibility with previous versions, and thus no reason to introduce one artificially through the library version numner.

With this, let's go back to the email-based workflow. We now have agreement over the general approach, so we can sort out the rest as usual.

Before posting the list, I'll try to cleanup the patch set. It's sort of unfortunate that it introduces `use_regular_socket` in one patch just to remove it again a few patches later. I *hope* I'll be able to resturtcure it without major efforts. If I'm not, remnants of `use_regular_socket` might remain.

mwilck committed:
multipathd: honor MULTIPATH_SOCKET_NAME environment variable

If multipathd is started via socket activation, it will obtain
sockets from systemd. The names of these sockets, and whether
the abstract and / or pathname socket is created, is configurable
in the systemd unit file.

Add support for passing a socket name via the environment, so that
it's possible to configure the socket name at runtime even without
socket activation. In this case, only this single socket will be created.
If creating the socket fails, multipathd startup will fail, too.

Signed-off-by: Martin Wilck <mwilck@suse.com>
Reviewed-by: Benjamin Marzinski <bmarzins@redhat.com>

mwilck commented:
I reverted back to the `#include` approach and added support for `MULTIPATH_SOCKET_NAME` in multipathd, too. I posted the new set to dm-devel.

mwilck commented:
Fixed with https://github.com/opensvc/multipath-tools/pull/119
